Simplify CI definitions

nicoonoclaste opened the pull request:
- [x] Simplify Windows image names
- [x] Automatically test against latest CPython
- [x] Automatically test against latest ~~& development~~ PyPy
- [x] Add PyPy 3.8

nicoonoclaste committed:
CI: Drop `-windowsservercore` in image names

pathunstrom commented:
Just because I'm curious: is there an actual semantic difference between the servercore version and just the python versions?

nicoonoclaste committed:
CI: Add images for PyPy 3.8 (dev version)

Unfortunately, the pypy images do not have an `rc` tag or such:
  https://github.com/docker-library/pypy/issues/65

nicoonoclaste committed:
CI: Require tests to pass on CPython 3.10

Apparently, the Bors configuration wasn't updated along the Cirrus.

nicoonoclaste committed:
CI: Test against the lastest stable CPython

This should “always” be a duplicate with tests for the latest
minor version, but it hedges against outdated CI definitions.

nicoonoclaste commented:
> Just because I'm curious: is there an actual semantic difference between the servercore version and just the python versions?

No difference whatsoever: both are shared tags that refers images for that particular CPython version and various Windows version; the one without `-windowsservercore` also refers to Linux images but that doesn't affect our Windows builds  ^^

nicoonoclaste committed:
CI: Automatically test against latest PyPy

Currently that's a duplicate with `pypy:3.7-slim` but will be helpful
one PyPy 3.8 is released.

nicoonoclaste committed:
CI/Windows: Fix image names

Apparently, the Windows images do not exist yet under the generic shared tags
for 3.7 and 3.8:  https://github.com/docker-library/python/issues/663

AstraLuma reviewed (approved):
bors r+
